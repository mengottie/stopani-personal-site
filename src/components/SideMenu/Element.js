import styled from 'styled-components';

import { white } from '../../styles/tokens';

const Element = styled.li`
    padding: 0.5em;
    list-style: none;
    color: ${white};
`;

export default Element;
