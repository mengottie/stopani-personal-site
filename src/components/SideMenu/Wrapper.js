import styled from 'styled-components';

import { darkestColor } from '../../styles/tokens';

const Wrapper = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    width: 70%;
    height: 100%;
    background-color: ${darkestColor};
    z-index: 150;
`;

export default Wrapper;
