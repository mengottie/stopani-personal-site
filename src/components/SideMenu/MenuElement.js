import React from 'react';
import PropTypes from 'prop-types';

import Element from './Element';

function MenuElement({ text }) {
    return(
        <Element>{text}</Element>
    )
}

MenuElement.propTypes = {
    text: PropTypes.string.isRequired
}

export default MenuElement;
