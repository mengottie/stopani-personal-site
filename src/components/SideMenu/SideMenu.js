import React from 'react';

import Wrapper from './Wrapper';
import MenuElement from './MenuElement';

function SideMenu() {
    return (
        <Wrapper>
            <ul>
                <MenuElement text={'Element 1'} />
                <MenuElement text={'Element 2'} />
                <MenuElement text={'Element 3'} />
            </ul>
        </Wrapper>
    )
}

export default SideMenu;
