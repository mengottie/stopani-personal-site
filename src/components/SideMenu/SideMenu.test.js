import React from 'react';
import { create } from 'react-test-renderer';

import SideMenu from './SideMenu';

jest.mock('./SideMenu', () => 'SideMenu');

describe('SideMenu', () => {
    let component;

    beforeEach(() => {
        component = create(<SideMenu />);
    })
    it('renders correctly', () => {
        expect(component.toJSON()).toMatchSnapshot();
    })
})
