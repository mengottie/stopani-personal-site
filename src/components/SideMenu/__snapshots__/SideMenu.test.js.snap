// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu renders correctly 1`] = `<SideMenu />`;
